core/src/main/scala/cr/pulsar/Consumer.scala

```diff
@@ -210,23 +219,31 @@ object Consumer {
     private case class OptionsImpl[F[_], E](
         initial: SubscriptionInitialPosition,
         logger: E => Topic.URL => F[Unit],
-        unsubscribeMode: UnsubscribeMode
+        unsubscribeMode: UnsubscribeMode,
+        autoAck: Boolean
     ) extends Options[F, E] {
       override def withInitialPosition(
           _initial: SubscriptionInitialPosition
       ): Options[F, E] =
         copy(initial = _initial)
+
       override def withLogger(_logger: E => (Topic.URL => F[Unit])): Options[F, E] =
         copy(logger = _logger)
+
       override def withUnsubscribeMode(
           _unsubscribeMode: UnsubscribeMode
       ): Options[F, E] =
         copy(unsubscribeMode = _unsubscribeMode)
+
+      override def withAutoAck(_autoAck: Boolean): Options[F, E] =
```

Comment:
I was thinking if we actually need a boolean param here. For the user interface we can just leave it parameter-less:

```
override def withAutoAck: Options[F, E] =
  copy(autoAck = true)
```

Reply:
Yes, I think I'd prefer that :+1: 

Since you're on it, you can also refactor the `UnsubscribeMode` to do the same. By default, `autoUnsubscribe` is `true`, and we can add a new option `withManualUnsubscribe` or something like that.